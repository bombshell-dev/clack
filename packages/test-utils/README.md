# `@clack/test-utils`

A bunch of useful test utiltiies for use inside clack.
