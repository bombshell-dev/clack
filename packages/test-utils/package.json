{
  "name": "@clack/test-utils",
  "private": true,
  "version": "1.0.0-alpha.1",
  "type": "module",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "types": "./dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bombshell-dev/clack.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/bombshell-dev/clack/issues"
  },
  "homepage": "https://github.com/bombshell-dev/clack/tree/main/packages/test-utils#readme",
  "files": ["dist", "CHANGELOG.md"],
  "author": {
    "name": "James Garbutt",
    "url": "https://github.com/43081j"
  },
  "license": "MIT",
  "packageManager": "pnpm@9.14.2",
  "scripts": {
    "build": "tsc",
    "prepack": "pnpm build"
  },
  "dependencies": {
  },
  "devDependencies": {
  }
}
