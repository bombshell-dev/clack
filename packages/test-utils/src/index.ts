export { MockReadable } from './mock-readable.js';
export { MockWritable } from './mock-writable.js';

export function nextTick(): Promise<void> {
	return new Promise<void>((resolve) => {
		setTimeout(resolve, 0);
	});
}
