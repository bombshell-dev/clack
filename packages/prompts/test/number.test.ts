import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest';
import * as prompts from '../src/index.js';
import { MockReadable, MockWritable } from './test-utils.js';

describe.each(['true', 'false'])('number (isCI = %s)', (isCI) => {
	let originalCI: string | undefined;
	let output: MockWritable;
	let input: MockReadable;

	beforeAll(() => {
		originalCI = process.env.CI;
		process.env.CI = isCI;
	});

	afterAll(() => {
		process.env.CI = originalCI;
	});

	beforeEach(() => {
		output = new MockWritable();
		input = new MockReadable();
	});

	afterEach(() => {
		vi.restoreAllMocks();
	});

	test('renders message', async () => {
		const result = prompts.number({
			message: 'foo',
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		await result;

		expect(output.buffer).toMatchSnapshot();
	});

	test('renders placeholder if set', async () => {
		const result = prompts.number({
			message: 'foo',
			placeholder: 5,
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(output.buffer).toMatchSnapshot();
		expect(value).toBe(0);
	});

	test('can cancel', async () => {
		const result = prompts.number({
			message: 'foo',
			input,
			output,
		});

		input.emit('keypress', 'escape', { name: 'escape' });

		const value = await result;

		expect(prompts.isCancel(value)).toBe(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('renders cancelled value if one set', async () => {
		const result = prompts.number({
			message: 'foo',
			input,
			output,
		});

		input.emit('keypress', '1', { name: '1' });
		input.emit('keypress', '2', { name: '2' });
		input.emit('keypress', '', { name: 'escape' });

		const value = await result;

		expect(prompts.isCancel(value)).toBe(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('renders submitted value', async () => {
		const result = prompts.number({
			message: 'foo',
			input,
			output,
		});

		input.emit('keypress', '1', { name: '1' });
		input.emit('keypress', '2', { name: '2' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(12);
		expect(output.buffer).toMatchSnapshot();
	});

	test('defaultValue sets the value but does not render', async () => {
		const result = prompts.number({
			message: 'foo',
			defaultValue: 10,
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(10);
		expect(output.buffer).toMatchSnapshot();
	});

	test('validation errors render and clear', async () => {
		const result = prompts.number({
			message: 'foo',
			validate: (val) => (val < 3 ? 'should be greater than 3' : undefined),
			input,
			output,
		});

		input.emit('keypress', '1', { name: '1' });
		input.emit('keypress', '', { name: 'return' });
		input.emit('keypress', '0', { name: '0' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(10);
		expect(output.buffer).toMatchSnapshot();
	});

	test('validation errors render and clear (using Error)', async () => {
		const result = prompts.number({
			message: 'foo',
			validate: (val) => (val < 3 ? new Error('should be greater than 3') : undefined),
			input,
			output,
		});

		input.emit('keypress', '1', { name: '1' });
		input.emit('keypress', '', { name: 'return' });
		input.emit('keypress', '0', { name: '0' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(10);
		expect(output.buffer).toMatchSnapshot();
	});

	test('placeholder is not used as value when pressing enter', async () => {
		const result = prompts.number({
			message: 'foo',
			placeholder: '  (hit Enter to use default)',
			defaultValue: 7,
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(7);
		expect(output.buffer).toMatchSnapshot();
	});

	test('0 when no value and no default', async () => {
		const result = prompts.number({
			message: 'foo',
			placeholder: '  (hit Enter to use default)',
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(0);
		expect(output.buffer).toMatchSnapshot();
	});
	
	test('warn when use invalid number value', async () => {
		const result = prompts.number({
			message: 'foo',
			placeholder: 1,
			input,
			output,
		});

		input.emit('keypress', 'a', { name: 'a' });
		input.emit('keypress', '', { name: 'return' });
		input.emit('keypress', '', { name: 'backspace' });
		input.emit('keypress', '1', { name: '1' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe(1);
		expect(output.buffer).toMatchSnapshot();
	});

	test('can be aborted by a signal', async () => {
		const controller = new AbortController();
		const result = prompts.number({
			message: 'foo',
			input,
			output,
			signal: controller.signal,
		});

		controller.abort();
		const value = await result;
		expect(prompts.isCancel(value)).toBe(true);
		expect(output.buffer).toMatchSnapshot();
	});
});
