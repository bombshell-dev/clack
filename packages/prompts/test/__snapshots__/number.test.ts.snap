// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`number (isCI = false) > 0 when no value and no default 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > can be aborted by a signal 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > can cancel 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31mâ– [39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > defaultValue sets the value but does not render 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > placeholder is not used as value when pressing enter 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m7[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > renders cancelled value if one set 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  12â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31mâ– [39m  
[90mâ”‚[39m  [9m[2m12[22m[29m
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > renders message 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > renders placeholder if set 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m5[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > renders submitted value 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  12â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m12[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > validate when use invalid number value 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m1[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  aâ–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  aâ–ˆ
[33mâ””[39m  [33mPlease input a valid number value[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  [7m1[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m1[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > validation errors render and clear (using Error) 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  1â–ˆ
[33mâ””[39m  [33mshould be greater than 3[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  10â–ˆ
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = false) > validation errors render and clear 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  1â–ˆ
[33mâ””[39m  [33mshould be greater than 3[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  10â–ˆ
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > 0 when no value and no default 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > can be aborted by a signal 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > can cancel 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31mâ– [39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > defaultValue sets the value but does not render 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > placeholder is not used as value when pressing enter 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m7[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > renders cancelled value if one set 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  12â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31mâ– [39m  
[90mâ”‚[39m  [9m[2m12[22m[29m
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > renders message 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > renders placeholder if set 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m5[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > renders submitted value 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  12â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m12[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > validate when use invalid number value 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m1[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  aâ–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  aâ–ˆ
[33mâ””[39m  [33mPlease input a valid number value[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  [7m1[27m[2m[22m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m1[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > validation errors render and clear (using Error) 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  1â–ˆ
[33mâ””[39m  [33mshould be greater than 3[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  10â–ˆ
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`number (isCI = true) > validation errors render and clear 1`] = `
[
  "<cursor.hide>",
  "[90mâ”‚[39m
[36mâ—†[39m  
[36mâ”‚[39m  [7m[8m_[28m[27m
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=2>",
  "<erase.line><cursor.left count=1>",
  "[36mâ”‚[39m  1â–ˆ",
  "<cursor.down count=2>",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33mâ–²[39m
[33mâ”‚[39m  1â–ˆ
[33mâ””[39m  [33mshould be greater than 3[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36mâ—†[39m  
[36mâ”‚[39m  10â–ˆ
[36mâ””[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32mâ—‡[39m  
[90mâ”‚[39m  [2m10[22m",
  "
",
  "<cursor.show>",
]
`;
